meorg_client/client.py

```diff
@@ -60,27 +72,35 @@ def _make_request(self, method: str, endpoint: str, data: dict=None, headers: di
         method = method.upper()
 
         if method not in VALID_METHODS:
-            raise Exception(f'Invalid method {method}')
+            raise Exception(f"Invalid method {method}")
 
         # GET/PUT requests have the data interpolated into the url
-        if method in ['GET', 'PUT']:
-            endpoint = endpoint.format(data)
+        if method in ["GET", "PUT"]:
+            endpoint = endpoint.format(**data)
 
-        url = f"{self.base_url}/{endpoint}"
+        url = urljoin(self.base_url, endpoint)
         all_headers = {**self.headers, **headers} if headers else self.headers
-        response = requests.request(method, url, data=data, headers=all_headers)
 
-        if response.status_code == 200:
+        # Update the client with the last response
+        self.last_response = requests.request(
+            method, url, data=data, headers=all_headers
+        )
 
+        if self.last_response.status_code in [200, 202]:
             # Return JSON if that's what it is (this should be the default)
-            if response.headers.get('Content-Type', str) == 'application/json':
-                return response.json()
-            
-            return response
+            if (
+                self.last_response.headers.get("Content-Type", str)
+                == "application/json"
+                and return_json == True
+            ):
+                return self.last_response.json()
```

Comment:
Consider moving strings to constant file

Reply:
Content types moved to constants.py